src/lime/media/AudioSource.hx

```diff
@@ -108,6 +111,16 @@ class AudioSource
 		return __backend.setLoops(value);
 	}
 
+	private function get_pitch():Float
+	{
+		return __backend.getPitch();
```

Comment:
That doesn't seem right. `__backend` should never be null, no matter what you're compiling to.

And if it did crash, wouldn't every other function crash too?

Reply:
>  Ah, based on the comment below, every other one does crash.

Wait, What comment??

Reply:
The one on `set_pitch()`. I assume if `get_pitch()` and `set_pitch()` both fail, they all will.

Reply:
I would say that just use the `if (__backend != null)` statement but idk if that's decent code lol

Reply:
> idk if that's decent code

I'd say it depends. Sometimes, null safety is worth adding the extra code. Other times, it's too much code for too little benefit. `AudioSource` has 17 different functions that reference `__backend`. If we never expect `__backend` to be null, is it worth the (tiny) performance hit?

And there's another angle on this: we might _want_ it to crash. There are situations where programmers [deliberately add crashes to a program](https://en.wikipedia.org/wiki/Assertion_(software_development\)#Assertions_for_run-time_checking). The program becomes a tiny bit slower _and_ a tiny bit more likely to crash.

So what's the point? The point is that crashes give information. They instantly tell you that something needs fixing, and (often) where the problem is.

Whereas if you bury the issue with a line like `if (__backend != null)`, and you could miss the problem if you aren't paying attention to audio.

---

Things would be different if we did expect `__backend` to be null sometimes. In that case, we'd want to check for it, and we'd have some kind of backup plan. But in reality, we never expect it to be null, and if it ever was null, that would be a huge problem. We'd want to know at once, so we could fix it.

Reply:
Ok, I think I might know what makes it crash

Reply:
>If we never expect __backend to be null, is it worth the (tiny) performance hit?

No, It isn't really worth it imo lol

Reply:
Ok, Set and get pitch are normal, you don't need to use if (__backend != null) So this is shitty code

Reply:
On the other hand, `NativeAudioSource` has `if (handle != null)` all over the place. I guess that's fine, since `handle` is expected to be null sometimes. (Dunno why they programmed it that way, but they did.)